core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            auto strVal = value.as<std::string>();
+            type = "int";
```

Comment:
On Wed, Sep 23, 2015 at 10:33 PM, Brett Camper <notifications@github.com>
wrote:

> In core/src/scene/sceneLoader.cpp
> <https://github.com/tangrams/tangram-es/pull/294#discussion_r40253896>:
>
> > @@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
> >      return out;
> >  }
> >
> > +StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
> > +    std::string type = "";
> > +    std::vector<UniformValue> uniformValues;
> > +    if (value.IsScalar()) { //float, int (bool), string (texture)
> > +        UniformValue uVal;
> > +        try {
> > +            uVal = value.as<float>();
> > +            type = "float";
> > +        } catch (const BadConversion& e) {
> > +            auto strVal = value.as<std::string>();
> > +            type = "int";
>
> I would prefer to not add support for `int` uniforms since they appear to
> not be supported in WebGL, and I'd like to keep parity so we don't have
> files that work on one platform but not the other (that's clearly pretty
> aspirational right now, but we can at least avoid making the gap wider :)
>

 That would be very reasonable -  I know that GLES does not support int for
vertex attributes either, though it seems WebGL does have int uniforms as
well:
http://webglfundamentals.org/webgl/lessons/webgl-shaders-and-glsl.html#uniforms